File: lib/models/ticket.dart
class Ticket {
  final String id;
  final String eventId;
  final String buyerName;
  final int seatNumber;
  final DateTime purchaseDate;
  final bool checkedIn;
  final List<String> addons;
  final Map<String, dynamic> answers;

  Ticket({
    required this.id,
    required this.eventId,
    required this.buyerName,
    required this.seatNumber,
    required this.purchaseDate,
    required this.checkedIn,
    required this.addons,
    required this.answers,
  });

  factory Ticket.fromJson(Map<String, dynamic> json) {
    return Ticket(
      id: json['_id']?.toString() ?? '',
      eventId: json['eventId'] is Map<String, dynamic>
          ? (json['eventId']['_id']?.toString() ?? '')
          : (json['eventId']?.toString() ?? ''),
      buyerName: json['buyerName']?.toString() ?? '',
      seatNumber: (json['seatNumber'] is num) ? (json['seatNumber'] as num).toInt() : 0,
      purchaseDate: DateTime.tryParse(json['purchaseDate']?.toString() ?? '') ?? DateTime.now(),
      checkedIn: json['checkedIn'] == true,
      addons: (json['addons'] is List) ? (json['addons'] as List).map((e) => e.toString()).toList() : <String>[],
      answers: (json['answers'] is Map<String, dynamic>) ? (json['answers'] as Map<String, dynamic>) : <String, dynamic>{},
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'eventId': eventId,
      'buyerName': buyerName,
      'seatNumber': seatNumber,
      'purchaseDate': purchaseDate.toIso8601String(),
      'checkedIn': checkedIn,
      'addons': addons,
      'answers': answers,
    };
  }
}

File: event_backend/models/Ticket.js
const mongoose = require("mongoose");

const TicketSchema = new mongoose.Schema(
  {
    buyerName: { type: String, required: true, trim: true },
    seatNumber: { type: Number, required: true, max: 100000 }, // Number with max
    checkedIn: { type: Boolean, default: false }, // Boolean
    purchaseDate: { type: Date, default: Date.now }, // Date
    notes: { type: [String], default: [] }, // Array
    payload: { type: mongoose.Schema.Types.Mixed, default: {} }, // JSON-like

    // Foreign key to Event (and Event references Organizer)
    eventId: {
      type: mongoose.Schema.Types.ObjectId,
      ref: "Event",
      required: true,
      index: true,
    },
  },
  { timestamps: true }
);

module.exports = mongoose.model("Ticket", TicketSchema);
